pymeasure/instruments/siglenttechnologies/siglent_spd1305x.py

```diff
@@ -36,10 +37,12 @@ def __init__(self, adapter, **kwargs):
             name="Siglent Technologies SPD1305X Power Supply",
             **kwargs
         )
-        voltage_limits = [0, 30]
-        current_limits = [0, 5]
+        voltages = [0, 30]
+        currents = [0, 5]
 
-        self.ch[1] = SPDChannel(self, 1, voltage_limits, current_limits)
+        self.channels = Instrument.ChannelCreator(SPDChannel, 1,
```

Comment:
That won't work. The channel creator has to be used at class level similarly to `control`. 
- You could move the variables at module level and then move the Creator at class level (similar to the Single Base Class).
- Or you move the creator (without the values) at class level and you set in this position `ch_1.voltage_range = xy`, similarly the current range. 
- Or you use add child in this position.

Reply:
Sorted